subsys/logging/log_core.c

```diff
@@ -544,6 +620,10 @@ char *log_strdup(const char *str)
 	struct log_strdup_buf *dup;
 	int err;
 
+	if (IS_ENABLED(CONFIG_LOG_INPLACE_PROCESS)) {
```

Comment:
- log_free(), which is public API still tries to free the result of log_strdup().
- log_is_strdup() is not needed when there is no pool.

Reply:
compiler figures out that they are not used (as they are used in log_msg functions which are not used) and removes them.